src/app/shared/licenses/licenses.service.ts

```diff
@@ -1,16 +1,20 @@
 import { Injectable } from '@angular/core';
 import { License } from './licenses.model';
 import { HttpClient } from '@angular/common/http';
-import { shareReplay } from 'rxjs';
+import { BehaviorSubject } from 'rxjs';
 
 @Injectable({
   providedIn: 'root',
 })
 export class LicensesService {
-  licenses$ = this.http.get<License[]>('/assets/licenses.json').pipe(
-    shareReplay(),
-  );
+
+  public licenses$: BehaviorSubject<License[]> = new BehaviorSubject<License[]>(null);
+
+  private readonly LICENSCE_URL = '/assets/licenses.json';
```

Comment:
> @fredx30 
> On another note though im intrested as to why the UPPER_CASE class constant is being rejected by codacy.

It's a readonly field. And all fields should conform to the camelCasing.

You could make it a `const`, and consts should be UPPERCASED.

So instead of:

```ts
export class LicensesService {
  // ...

  private readonly LICENCE_URL = '/assets/licenses.json';

  // ... use of `this.LICENSE_URL`
}
```

You do:

```ts
const LICENCE_URL = '/assets/licenses.json';

export class LicensesService {

  // ... use of `LICENSE_URL`

}
```

Reply:
Good workaround. That puts it at the top of the document aswell.